app/graphql/mutations/create_signup.rb

```diff
@@ -5,10 +5,12 @@ class CreateSignup < BaseMutation
     null true
 
     argument :event_id, ID, required: true
+    argument :user_email, String, required: false
 
-    def resolve(event_id:)
+    def resolve(event_id:, user_email: nil)
```

Comment:
This resolver was originally designed to leverage the current session user. Since the current user would always have permission to sign themselves up for an event, it was safe.

However, adding an optional email argument without any authorisation, would mean that any user would be able to sign any other user up for an event.

Alternatively, separating the mutations would make more sense here imo:
- They are different operations and require different inputs. i.e. email is not actually optional in the context of an admin adding a person to an event.
- They require different permissions/roles. i.e. only admins should be able to access this mutation.
- The clear separation will help maintainability and dev onboarding